### PR TITLE
Fix examples: return -> returnValue, add interface for run-blockchain

### PR DESCRIPTION
**Pull request by hugo-dc**
Found a couple of issues while executing the examples

### CODE REVIEW

File: examples/run-blockchain/index.ts [hunk 1]
```diff
@@ -11,6 +11,15 @@ const BlockHeader = require('ethereumjs-block/header.js')
 const testData = require('./test-data')
 const level = require('level')
 
+interface IBlockchainHead {
```
Comment (s1na) [inline examples/run-blockchain/index.ts]: Did you get an error without this interface? For reference, `ethereumjs-block` has been migrated to typescript, but is pending a release https://github.com/ethereumjs/ethereumjs-block/pull/74. Afterwards we could just use the `Block` type.

File: examples/run-blockchain/index.ts [hunk 2]
```diff
@@ -39,10 +48,12 @@ async function main() {
...
   await vm.runBlockchain(blockchain)
 
-  const blockchainHead = await promisify(vm.blockchain.getHead.bind(vm.blockchain))()
+  const blockchainHead: IBlockchainHead = await promisify(
+    vm.blockchain.getHead.bind(vm.blockchain),
+  )()
 
   console.log('--- Finished processing the BlockChain ---')
-  console.log('New head:', '0x' + blockchainHead.hash().toString('hex'))
+  console.log('New head:', '0x' + blockchainHead.header.hash().toString('hex'))
```
Comment (s1na) [inline examples/run-blockchain/index.ts]: Is this for clarification purposes? `blockchainHead.hash()` should also work right?

File: examples/run-code-browser/index.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ vm.runCode({
   gasLimit: new BN(0xffff),
 })
   .then(results => {
-    console.log('Returned : ' + results.return.toString('hex'))
+    console.log('Returned : ' + results.returnValue.toString('hex'))
```
Comment (s1na) [inline examples/run-code-browser/index.js]: For reference, `return` was changed to `returnValue` in #551.

File: examples/run-blockchain/index.ts (hunk 1 above)
Comment (hugo-dc) [reply examples/run-blockchain/index.ts]: Yes, without the interface I get this error:

```
/home/hugo/.nvm/versions/node/v10.16.0/lib/node_modules/ts-node/src/index.ts:245
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
index.ts:58:50 - error TS2339: Property 'header' does not exist on type '{}'.

58   console.log('New head:', '0x' + blockchainHead.header.hash().toString('hex'))
                                                    ~~~~~~
```

`blockchainHead.hash()` doesn't work either:

```
TSError: ⨯ Unable to compile TypeScript:
index.ts:58:50 - error TS2339: Property 'hash' does not exist on type '{}'.

58   console.log('New head:', '0x' + blockchainHead.hash().toString('hex'))
```

File: examples/run-blockchain/index.ts (hunk 2 above)
Comment (hugo-dc) [reply examples/run-blockchain/index.ts]: Yes it works if `hash` is added to the interface: 

```
interface IBlockchainHead {
  _common?: any
  transactions?: any
  uncleHeaders?: any
  _inBlockchain?: any
  txTrie?: any
  header?: any
  hash?: any
}
```

Let me know if it should be added, or if there is a better way than using the interface to fix this issue.

**Comment by alcuadrado** [comment]
Hey @hugo-dc, thanks for this fix.

The errors you are getting is because you are using TS 3.5.x, and we (because of an older `node_modules`) are/were using 3.4.x. We have TS dependency declared with a `^`, but I recently learned that they introduce breaking changes in minor versions, so we should declare it with a `~`.

In particular, 3.5 changed the inference a little to make the type system stricter. Previously, `{}` (which is almost equivalent to `any`) was inferred in some cases. Now those inferences resolve to `unknown` which in many cases end up in a compilation error. In this case, a call to `promisify` was the problem.

I created [this other branch](https://github.com/ethereumjs/ethereumjs-vm/tree/rename-return-to-returnValue) to confirm it. I also renamed a `return` in `tests/VMTestsRunner.js`.

**Comment by alcuadrado** [comment]
I'm closing this issue as #604 was merged and fixed this.

Thanks @hugo-dc for your collaboration!

**Comment by hugo-dc** [comment]
@alcuadrado thanks for the explanation and the fix.

**Pull request closed without merging**